node/request.js

```diff
@@ -329,7 +334,7 @@ TChannelRequest.prototype.shouldRetryError = function shouldRetryError(err) {
                 return true;
 
             case 'Timeout':
-                return !!self.options.retryFlags.onTimeout;
+                return true;
```

Comment:
Respect `!!self.options.retryFlags.onTimeout`; dont set this to true.

Sorry; didnt communicate properly.